test/e2e_env/kubernetes/observability/applications_metrics.go

```diff
@@ -189,4 +226,41 @@ func ApplicationsMetrics() {
 		// metric from envoy
 		Expect(stdout).To(ContainSubstring("envoy_server_concurrency"))
 	})
+
+	It("should return envoy filtered metrics", func() {
+		// given
+		podName, err := PodNameOfApp(env.Cluster, "test-server-filter", namespace)
+		Expect(err).ToNot(HaveOccurred())
+		podIp, err := PodIPOfApp(env.Cluster, "test-server-filter", namespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		// when
+		stdout, _, err := env.Cluster.Exec(namespace, podName, "test-server-filter",
+			"curl", "-v", "-m", "3", "--fail", "http://"+net.JoinHostPort(podIp, "1234")+"/metrics")
```

Comment:
Can we move the:

```
"http://"+net.JoinHostPort(podIp, "1234")+"/metrics"
```

part to a function, seems like it's called the same way from multiple places.

Reply:
Reading the test might be not that easy because you will have to go and check what was the path/port. I've changed also the port to check a more dynamic configuration.